receiver/googlecloudmonitoringreceiver/receiver.go

```diff
@@ -5,51 +5,203 @@ package googlecloudmonitoringreceiver // import "github.com/open-telemetry/opent
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/multierr"
 	"go.uber.org/zap"
+
+	"google.golang.org/api/iterator"
+	"google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudmonitoringreceiver/internal"
+
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
+	
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
 )
 
 type monitoringReceiver struct {
-	config *Config
-	logger *zap.Logger
-	cancel context.CancelFunc
+	config         *Config
+	logger         *zap.Logger
+	client         *monitoring.MetricClient
+	metricsBuilder *internal.MetricsBuilder
+	startOnce      sync.Once
 }
 
 func newGoogleCloudMonitoringReceiver(cfg *Config, logger *zap.Logger) *monitoringReceiver {
 	return &monitoringReceiver{
-		config: cfg,
-		logger: logger,
+		config:         cfg,
+		logger:         logger,
+		metricsBuilder: internal.NewMetricsBuilder(logger),
 	}
 }
 
-func (m *monitoringReceiver) Scrape(ctx context.Context) (pmetric.Metrics, error) {
-	// Dummy use to fix lint errors
-	ctx.Deadline()
-	metrics := pmetric.NewMetrics()
-	m.logger.Debug("Scrape metrics ")
+func (mr *monitoringReceiver) Start(ctx context.Context, _ component.Host) error {
+	var startErr error
+	mr.startOnce.Do(func() {
```

Comment:
Yes, I implemented a retry function with a 5-second delay.

Reply:
You should leave retry logic to the underlying collector framework.  I just meant to say that a sync.Once won't work here because we may need this to run again if it failed the first time.

Reply:
Got it, Updated Now